x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_tables.tsx

```diff
@@ -260,16 +267,27 @@ export const RulesTables = React.memo<RulesTableProps>(({ selectedTab }) => {
       {isTableEmpty && <PrePackagedRulesPrompt />}
       {isDeleteConfirmationVisible && (
         <EuiConfirmModal
-          title={i18n.DELETE_CONFIRMATION_TITLE}
+          title={
+            rulesToDeleteCount === 1
+              ? i18n.SINGLE_DELETE_CONFIRMATION_TITLE
+              : i18n.BULK_DELETE_CONFIRMATION_TITLE
+          }
           onCancel={handleDeletionCancel}
           onConfirm={handleDeletionConfirm}
           confirmButtonText={i18n.DELETE_CONFIRMATION_CONFIRM}
           cancelButtonText={i18n.DELETE_CONFIRMATION_CANCEL}
           buttonColor="danger"
           defaultFocusedButton="confirm"
-          data-test-subj="allRulesDeleteConfirmationModal"
+          data-test-subj="deleteRulesConfirmationModal"
         >
-          <p>{i18n.DELETE_CONFIRMATION_BODY}</p>
+          <FormattedMessage
+            id="xpack.securitySolution.detectionEngine.components.allRules.deleteConfirmationModalBody"
+            defaultMessage='This action will delete {rulesToDeleteCount, plural, one {the chosen rule} other {{rulesToDeleteCountStrong} rules}}. Click "Confirm" to continue.'
```

Comment:
nit: Might want to rework this text as part of the other i18n comment around the confirm button text matching the action (`delete`) in this case.